presto-main/src/main/java/com/facebook/presto/operator/TableScanOperator.java

```diff
@@ -254,10 +255,15 @@ public Page getOutput()
 
             // update operator stats
             long endCompletedBytes = source.getCompletedBytes();
+            long endCompletedPositions = source.getCompletedPositions();
+            if (endCompletedPositions < 0) {
+                endCompletedPositions = completedPositions + page.getPositionCount();
```

Comment:
Trying to understand -- why would statement ever be true?

Reply:
Connectors that don't implement the new method will use the default implementation that returns -1.